packages/state-transition/src/cache/epochCache.ts

```diff
@@ -736,9 +768,62 @@ export class EpochCache {
     return isAggregatorFromCommitteeLength(committee.length, slotSignature);
   }
 
-  addPubkey(index: ValidatorIndex, pubkey: Uint8Array): void {
-    this.pubkey2index.set(pubkey, index);
-    this.index2pubkey[index] = bls.PublicKey.fromBytes(pubkey, CoordType.jacobian); // Optimize for aggregation
+  getPubkey(index: ValidatorIndex): PublicKey {
```

Comment:
add comment why we only get from `finalizedIndex2pubkey`, or change method name to `getFinalizedPubkey()`

Reply:
> add comment why we only get from `finalizedIndex2pubkey`, or change method name to `getFinalizedPubkey()`

Will add comment. 

I did an [analysis](https://hackmd.io/@adrninistrator1/SkHmz972n) on the entire consensus spec and Lodestar codebase and concluded there is no place in the client that needs to enquire for unfinalized pubkey so I think we can safely assume by pubkey we are referring to finalized pubkey. This is also the reason why there is no `unfinalizedIndex2Pubkey`.